cpp/unreal_plugins/SpEngine/Source/SpEngine/LegacyService.h

```diff
@@ -0,0 +1,337 @@
+            SP_ASSERT(task_);
+            return task_->getStepInfo();
+        });
+
+        entry_point_binder->bind_func_wrapped("legacy_service", "reset_agent", [this]() -> void {
+            SP_ASSERT(agent_);
+            agent_->reset();
+        });
+
+        entry_point_binder->bind_func_wrapped("legacy_service", "reset_task", [this]() -> void {
+            SP_ASSERT(task_);
+            task_->reset();
+        });
+
+        entry_point_binder->bind_func_wrapped("legacy_service", "is_agent_ready", [this]() -> bool {
+            SP_ASSERT(agent_);
+            return agent_->isReady();
+        });
+
+        entry_point_binder->bind_func_wrapped("legacy_service", "is_task_ready", [this]() -> bool {
+            SP_ASSERT(task_);
+            return task_->isReady();
+        });
+
+        entry_point_binder->bind_func_wrapped("legacy_service", "get_random_points",
+            [this](const int& num_points) -> std::vector<double> {
+                SP_ASSERT(nav_mesh_);
+                return nav_mesh_->getRandomPoints(num_points);
+        });
+
+        entry_point_binder->bind_func_wrapped("legacy_service", "get_random_reachable_points_in_radius",
+            [this](const std::vector<double>& initial_points, const float& radius) -> std::vector<double> {
+                SP_ASSERT(nav_mesh_);
+                return nav_mesh_->getRandomReachablePointsInRadius(initial_points, radius);
+        });
+
+        entry_point_binder->bind_func_wrapped("legacy_service", "get_paths",
+            [this](const std::vector<double>& initial_points, const std::vector<double>& goal_points) -> std::vector<std::vector<double>> {
+                SP_ASSERT(nav_mesh_);
+                return nav_mesh_->getPaths(initial_points, goal_points);
+        });
+	}
+
+    ~LegacyService()
+    {
+        // If an object of this class is destroyed in the middle of simulation for some reason, raise an error.
+        // We expect worldCleanUpEvenHandler(...) to be called before ~LegacyService().
+        SP_ASSERT(!world_begin_play_handle_.IsValid());
+
+        FWorldDelegates::OnWorldCleanup.Remove(world_cleanup_handle_);
+        FWorldDelegates::OnPostWorldInitialization.Remove(post_world_initialization_handle_);
+
+        world_cleanup_handle_.Reset();
+        post_world_initialization_handle_.Reset();
+    }
+
+    void postWorldInitializationHandler(UWorld* world, const UWorld::InitializationValues initialization_values)
+    {
+        SP_LOG_CURRENT_FUNCTION();
+        SP_ASSERT(world);
+
+        #if WITH_EDITOR // defined in an auto-generated header
+            bool world_is_ready = world->IsGameWorld();
+        #else
+            bool world_is_ready = world->IsGameWorld() && GEngine->GetWorldContextFromWorld(world);
+        #endif
+
+        if (world_is_ready) {
+
+            std::string scene_id = "apartment_0000";
```

Comment:
Is this hard-coding intended? It is fine if it is (for now), but it looks like it might have been for debugging.

Reply:
It's not needed.